src/author/author.repository.ts

```diff
@@ -1,56 +1,61 @@
 import { Injectable, NotFoundException } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository } from 'typeorm';
 import { CreateAuthorDto } from './dto/create-author.dto';
 import { UpdateAuthorDto } from './dto/update-author.dto';
+import { Author } from './entities/author.entity';
 
 @Injectable()
 export class AuthorRepository {
-  private authors = [];
+  constructor(
+    @InjectRepository(Author)
+    private authorRepository: Repository<Author>,
+  ) {}
 
-  create(createAuthorDto: CreateAuthorDto) {
-    const newAuthor = { id: this.authors.length + 1, ...createAuthorDto };
-    this.authors.push(newAuthor);
-    return newAuthor;
+  async create(createAuthorDto: CreateAuthorDto) {
+    const author = this.authorRepository.create(createAuthorDto);
+    return await this.authorRepository.save(author);
   }
-  findAll(search?: string) {
+
+  async findAll() {
+    return await this.authorRepository.createQueryBuilder('author').getMany();
+  }
+
+  async findAllSearch(search?: string) {
     if (search) {
```

Comment:
you can write like this: 
const query = this.authorRepository
        .createQueryBuilder('author')
if(search){
query..where('author.firstName LIKE :search', { search: `%${search}%` })
        .orWhere('author.lastName LIKE :search', { search: `%${search}%` })
        .orWhere('author.biography LIKE :search', { search: `%${search}%` })
}
return await query.getMany()